### PR TITLE
Add pyglet recipe

### LINKED ISSUE #1029: Package pyglet
This seems like a super neat library that would be great to package, which I only now have heard of. Basically a Python interface to the OpenGL library. Looks like it can be used as a `matplotlib` backend too.

### PR DESCRIPTION
**Pull request by mwcraig**
Pinging @r1chardj0n3s @silvester747 @bitcraft in case they'd like to be added as a maintainer to the pyglet builder on conda-forge, a library of community-built conda packages. The maintenance burden here is light: all testing and releases are built and deployed automatically with CI. Changes to how the package is being built can be controlled by changing the recipe which will be located at https://github.com/conda-forge/pyglet-feedstock shortly after this PR is finished and merged. Maintaining the feedstock boils down to updating the version number and checksum when a new release is made on pypi. If you aren't interested in helping maintain the build, that's entirely fine too.

Ping @jakirkham -- I've added you as a maintainer too. Closes #1029

### CODE REVIEW

**Comment by mwcraig** [comment]
@jakirkham -- this is failing because it is trying to get a display on linux, and circle-ci does not (of course) have one. Should I drop that import test (part of a font import)? Is there a better solution?

**Comment by jakirkham** [comment]
Sorry was out. Something to try is `xvfb`, which is a virtual display on Linux. We used it with `qtconsole` with some success. To do this add `yum_requirements.txt` in to the recipe directory and add `xorg-x11-server-Xvfb` in it. There might be some tweaking to do, but you can run your import tests via a mechanism like [this]( https://github.com/conda-forge/qtconsole-feedstock/blob/421859b54e7ead2c2193971aefda86c8e55dbc57/recipe/meta.yaml#L38 ).

**Comment by jakirkham** [comment]
Probably can dump all your import tests into a Python file and then run them using the same mechanism outlined above.

**Comment by mwcraig** [comment]
Thanks for the information, will try to implement in the morning.

**Comment by jakirkham** [comment]
Sounds good. Feel free to ask questions if you get stuck.

**Comment by mwcraig** [comment]
@jakirkham -- can you help me out with running the import tests from a python file? See the latest circleci fail, for example.

**Comment by mwcraig** [comment]
Given the latest circleci fail I'm now inclined to simply skip, on linux, all of the tests I've separated out. I think they require actual graphics hardware to really work.

I'll wait for the other CI tests to run and try to fix any remaining issues in one more commit.

**Comment by jakirkham** [comment]
Yep, I agree with you. It was worth a try, but it is clearly not going to work.

Would it be alright if we remove the `yum_requirements.txt` and get rid of the Linux special casing since we are not going down that path?

Also, we could try to tweak the import file that we have and try something like [this]( https://github.com/conda-forge/python-drmaa-feedstock/blob/a1f223818f2a2fed72bd9b1e497ae6a374450728/recipe/run_test.py ). Should be pretty easy. Basically counting the original exception as a success as it demonstrates the modules was there and threw this error.

**Comment by jakirkham** [comment]
That will probably help on Travis CI too.

**Comment by mwcraig** [comment]
> Also, we could try to tweak the import file that we have and try something like this. Should be pretty easy. Basically counting the original exception as a success as it demonstrates the modules was there and threw this error.

@jakirkham -- were you thinking of doing this for each of the failing packages? It would be easy enough to implement a check that the error generated for each failed import is of the "correct" type.

**Comment by mwcraig** [comment]
Decided to make one last attempt along the lines @jakirkham suggested....that failed, so I'm going to pull out the yum requirements and the special handling of some of the imports. The Windows test pass, so there is some checking that the packages are being built.

File: recipes/pyglet/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    - pyglet.media  # [win]
+    - pyglet.media.drivers  # [win]
+    - pyglet.media.drivers.directsound  # [win]
+    - pyglet.window  # [win]
+    - pyglet.text  # [win]
+    - pyglet.text.formats  # [win]
+
+about:
+    home: http://pyglet.readthedocs.org/en/pyglet-1.2-maintenance/
+    license: BSD License
```
Comment (jakirkham) [inline recipes/pyglet/meta.yaml]: Could you please specify the number of clauses and drop "License"?

File: recipes/pyglet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    summary: 'Cross-platform windowing and multimedia library'
+
+extra:
+    recipe-maintainers:
+        - mwcraig
+        - jakirkham
+        - r1chardj0n3s
+        - silvester747
+        - bitcraft
+
```
Comment (jakirkham) [inline recipes/pyglet/meta.yaml]: nit: please drop extra newline.

File: recipes/pyglet/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/pyglet/meta.yaml]: Looks like BSD 3-Clause.

**Comment by jakirkham** [comment]
Fair enough. One or two minor easy changes and then would be happy to merge.

File: recipes/pyglet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    - pyglet.media  # [win]
+    - pyglet.media.drivers  # [win]
+    - pyglet.media.drivers.directsound  # [win]
+    - pyglet.window  # [win]
+    - pyglet.text  # [win]
+    - pyglet.text.formats  # [win]
+
+about:
+    home: http://pyglet.readthedocs.org/en/pyglet-1.2-maintenance/
+    license: BSD 3-Clause License
```
Comment (jakirkham) [inline recipes/pyglet/meta.yaml]: Normally we drop "License" too as it is already clear, but it is not a big deal either way.

**Comment by jakirkham** [comment]
Thanks @mwcraig.

File: recipes/pyglet/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/pyglet/meta.yaml]: SGTM

**Comment by mwcraig** [comment]
Thanks for your help, @jakirkham

**Comment by jakirkham** [comment]
Sure. Sorry for the wild goose chase on the import tests. We can take a closer look in the feedstock.

**Comment by bitcraft** [comment]
Pretty exciting for pyglet to be added here.  I'm not sure what help I could be...I'm not really familiar with automated builds and what not.  I'm going to look everything over.  @mwcraig thanks for your work so far (and anyone else!).

**Comment by mwcraig** [comment]
@bitcraft - the only effort is updating the version number and md5 hash in this repo when a new a release is made: https://github.com/conda-forge/pyglet-feedstock

The usual practice is to open a PR in that repo, which triggers builds on all platforms. Once those pass, merge the PR and the package is built again and uploaded. (You do have commit rights on that repo, as do I and a couple more conda-forge people.)

Even just opening an issue there when you have a new release will be helpful in keeping the package up to date.

**Pull request merged**